### PR TITLE
Add queryRecentChanges API for time-range memory feed

### PR DESCRIPTION
**Pull request by marcelsamyn**
## Summary
Adds a new `queryRecentChanges` API endpoint that returns a time-range "what's new in memory" feed for daily digests and dashboards. The feed includes active personal claims and nodes added/updated within a time window, with labels and provenance pre-fetched to avoid N+1 queries.

## Key Changes

- **New query function** (`src/lib/query/recent-changes.ts`): Core logic that queries claims and nodes changed within `[since, until]`, with intelligent filtering:
  - Returns only active personal claims (excludes superseded and reference-scoped claims)
  - Separates nodes into "added" (created in window) and "updated" (touched by a claim in window)
  - Excludes structural node types (`Temporal`, `Atlas`, `AssistantDream`) by default unless explicitly requested
  - Joins labels and source metadata to avoid consumer N+1 fan-out
  - Supports optional `nodeTypes` filter to narrow both claims and nodes

- **Schema definitions** (`src/lib/schemas/query-recent-changes.ts`): Request/response types with validation:
  - `QueryRecentChangesRequest`: userId, since (required), until (optional, defaults to now), limit (1-500, default 100), nodeTypes filter
  - `QueryRecentChangesResponse`: claims, nodes, and sources arrays with change metadata
  - `ChangeKind` enum: "added" vs "updated" to distinguish fresh items from touched existing ones

- **HTTP route** (`src/routes/query/recent-changes.ts`): POST endpoint that parses and validates the request, delegates to the query function, and validates the response

- **SDK integration** (`src/sdk/memory-client.ts`, `src/sdk/index.ts`): Exposes `queryRecentChanges()` method on the MemoryClient with full JSDoc

- **Comprehensive test suite** (`src/lib/query/recent-changes.test.ts`, `src/query-recent-changes-route.test.ts`):
  - Integration test with real database setup/teardown covering edge cases: superseded claims, reference-scoped claims, stale claims, structural nodes, and nodeTypes filtering
  - Route handler test validating request parsing, default limit application, and schema validation

## Notable Implementation Details

- Uses `GREATEST(createdAt, updatedAt)` to determine claim change time, ordering by most recent change
- Merges added and updated nodes from separate queries, then sorts and limits the combined result to avoid double-counting
- Filters claims by both subject and object node types using a correlated EXISTS subquery to avoid self-join type inference issues
- Derives source titles from metadata with fallback chain (title → filename → null)
- Handles empty time ranges gracefully (since > until returns empty feed)

https://claude.ai/code/session_015Se2dNXXAv6pUCx1kyGPWu

### CODE REVIEW

File: src/lib/query/recent-changes.ts [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        claimChangedInWindow,
+      ),
+    )
+    .leftJoin(nodeMetadata, eq(nodeMetadata.nodeId, nodes.id))
+    .where(
+      and(eq(nodes.userId, userId), lt(nodes.createdAt, since), nodeTypeFilter),
+    )
+    .groupBy(nodes.id, nodes.nodeType, nodes.createdAt, nodeMetadata.label)
+    .orderBy(desc(lastClaimChange))
+    .limit(limit);
```
[the remark this answers is not in the document]
Comment (marcelsamyn) [reply src/lib/query/recent-changes.ts]: Done in 6f16a09. `updatedNodeRows` now resolves candidate node ids + their most-recent change time in-memory from the already `limit`-capped `claimRows`, then looks them up by primary key via `inArray` — dropping the `subjectNodeId = id OR objectNodeId = id` join along with its `groupBy`/aggregate. Behavior is unchanged: any claim touching a node whose type matches `nodeTypes` already passes the claim-level filter, so the candidate set is identical. Confirmed by the Postgres integration test (same ordering and `nodeTypes` results).

---
_Generated by [Claude Code](https://claude.ai/code)_

File: src/lib/query/recent-changes.ts [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+    title: z.string().min(1).optional(),
+    filename: z.string().min(1).optional(),
+  })
+  .catchall(z.unknown());
+
+function deriveSourceTitle(metadata: unknown): string | null {
+  const parsed = sourceTitleMetadataSchema.safeParse(metadata ?? {});
+  if (!parsed.success) return null;
+  return parsed.data.title ?? parsed.data.filename ?? null;
+}
```
[the remark this answers is not in the document]
Comment (marcelsamyn) [reply src/lib/query/recent-changes.ts:65]: Done in 6f16a09. The metadata schema now accepts any string and `deriveSourceTitle` trims + enforces non-empty, so an empty `title` falls through to `filename` instead of failing the whole parse.

---
_Generated by [Claude Code](https://claude.ai/code)_
